src/qiboconnection/models/job_result.py

```diff
@@ -43,16 +44,24 @@ def __post_init__(self) -> None:
         Decodes data from the http_response provided at instance creation, and uses that info to build the self.data
         attribute.
         """
-
-        if self.job_type == JobType.CIRCUIT:
-            self.data = decode_results_from_circuit(self.http_response)
-            return
-        if self.job_type == JobType.QPROGRAM:
-            self.data = decode_results_from_qprogram(self.http_response)
-            return
-        if self.job_type == JobType.VQA:
-            self.data = decode_results_from_qprogram(self.http_response)
+        try:
+            self.data = decompress_any(json.loads(self.http_response)["data"])
             return
+        except Exception as ex:  # pylint: disable=broad-exception-caught  # delete ASAP
+            logger.warning(
```

Comment:
Looks good, but you'll only have the Exception description.

Maybe you want to print the traceback too, to help with debugging?